pkg/proc/breakpoints.go

```diff
@@ -163,6 +168,22 @@ type returnBreakpointInfo struct {
 // CheckCondition evaluates bp's condition on thread.
 func (bp *Breakpoint) CheckCondition(thread Thread) BreakpointState {
 	bpstate := BreakpointState{Breakpoint: bp, Active: false, Internal: false, CondError: nil}
+	bpstate = bp.checkCond(bpstate, thread)
+	// Update the breakpoint hit counts.
+	if bpstate.Breakpoint != nil && bpstate.Active {
+		if g, err := GetG(thread); err == nil {
+			bpstate.HitCount[g.ID]++
+		}
+		bpstate.TotalHitCount++
+	}
+	//Check hit condition if this is a user breakpoint.
+	if bpstate.Active && !bpstate.Internal {
+		bpstate = bp.checkHitCond(bpstate, thread)
+	}
+	return bpstate
+}
+
+func (bp *Breakpoint) checkCond(bpstate BreakpointState, thread Thread) BreakpointState {
```

Comment:
I think the correct receiver of this function and of checkHitCond is `*BreakpointState` and then they don't return anything.

Reply:
Done